EFCache/InMemoryCache.cs

```diff
@@ -165,11 +170,35 @@ public int Count
             get { return _cache.Count; }
         }
 
+        public IEnumerable<string> EntitySetsInCache
+        {
+            get
+            {
+                return _entitySetToKey.Select(x => x.Key);
+            }
+        }
+
+        public int GetCacheSize()
+        {
+            lock (_cache)
+            {
+                var bf = new BinaryFormatter();
+                var ms = new MemoryStream();
+
+                bf.Serialize(ms, this);
+
+                var array = ms.ToArray();

```

Comment:
Now the cache takes 3 times more memory than it originally did - actual dictionary containing data, memory stream and the array. You could probably get to 2x by returning `ms.Position` instead of creating a new array.